### DISCUSSION TITLE
Is it possible to explore application protocols in MOSAIC?

### DISCUSSION THREAD

**Discussion by LDVictor** [eclipse/mosaic#63] [Q&A]
Hi guys. 

I'm a beginner in Eclipse MOSAIC and I noticed that MOSAIC uses a structure of "interactions" for its applications and does not define protocols in application layer. My doubt is: there is some possibility to adapt the interactions structure of MOSAIC applications to work with application protocols like BSM, CoAP or MQTT?

**Comment by kschrab** [comment]
Hi, the interactions in MOSAIC are only used to exchange data between the single simulators coupled to the runtime infrastructure. This has nothing to do with how V2X messages are exchanged and which layers of the communication stack are modeled. The communication stack itself (everything from the physical up to the application layer) is modeled in the communication simulator, e.g. OMNeT++ or ns-3. Information about received V2X messages is send to the application simulator using interactions, where you then can work with those messages at application layer. There you could indeed integrate further application protocols such as MQTT. We do quite similar things in some of our research projects.
